Library/Homebrew/utils/shebang.rb

```diff
@@ -34,13 +34,17 @@ def initialize(regex, max_length, replacement)
     # @api public
     sig { params(rewrite_info: RewriteInfo, paths: T::Array[T.any(String, Pathname)]).void }
     def rewrite_shebang(rewrite_info, *paths)
+      found = T.let(false, T::Boolean)
       paths.each do |f|
         f = Pathname(f)
         next unless f.file?
         next unless rewrite_info.regex.match?(f.read(rewrite_info.max_length))
 
         Utils::Inreplace.inreplace f.to_s, rewrite_info.regex, "#!#{rewrite_info.replacement}"
+        found = true
       end
+
+      raise "No matching files found to rewrite shebang" unless found
```

Comment:
```suggestion
      found = paths.all? do |f|
        f = Pathname(f)
        next false unless f.file?
        next true unless rewrite_info.regex.match?(f.read(rewrite_info.max_length))

        Utils::Inreplace.inreplace f.to_s, rewrite_info.regex, "#!#{rewrite_info.replacement}"
        true
      end

      raise "No matching files found to rewrite shebang" unless found
```
@Rylan12 maybe something like this then?

`#all?` probably doesn't work though, since I suspect it is greedy (bails out of the iterator as soon as it sees `false`)

Reply:
Yeah, that looks like it would work (I think you're right about `#all?` though so we might have to write that ourselves).

We also might want to change the error message since it's not necessarily that _no files_ were found just that one file wasn't.